trieste/acquisition/sampler.py

```diff
@@ -19,20 +19,23 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
+from collections.abc import Callable
 
 import tensorflow as tf
 import tensorflow_probability as tfp
+from gpflux.layers.basis_functions import RandomFourierFeatures
 from scipy.optimize import bisect
 
+from ..data import Dataset
 from ..models import ProbabilisticModel
 from ..type import TensorType
 from ..utils import DEFAULTS
 
 
-class Sampler(ABC):
+class DiscreteSampler(ABC):
```

Comment:
In future we'll have to think about the compatibility issues involved with renaming classes. For now I expect it's ok (though if we think anyone might already be using this, we could always add a `Sampler = DiscreteSampler` alias).

Reply:
None every uses Sampler, just the subclasses. But I get your point!

Reply:
The name of the type has changed though, so type annotations would need to be updated. Projects which consume `trieste` may also have added their own custom subclasses which would break.

Reply:
Understood :) I think more carefully before making these sort of changes in the future!